### ISSUE TITLE
Boost.X stylization in library pagelets

### ISSUE THREAD

**Issue by joaquintides** [boostorg/website-v2#1043]
The Libraries [home page](https://www.stage.boost.cppalliance.org/libraries) names libraries in exactly the same way as the [old website](https://www.boost.org/doc/libs/), some relevant examples follow:

* Accumulators
* Container Hash
* Multi-Array
* Functional/Factory
* Functional/Overloaded Function
* STL_interfaces
* Property Map (Parallel)

Now, within each library pagelet, the "Boost.X" stylization is used by simply prepending "Boost." to the previous names:

* [Boost.Accumulators](https://www.stage.boost.cppalliance.org/libraries/accumulators/boost-1-84-0/)
* [**Boost.Container Hash**](https://www.stage.boost.cppalliance.org/libraries/container_hash/boost-1-84-0/)
* [**Boost.Multi-Array**](https://www.stage.boost.cppalliance.org/libraries/multi_array/boost-1-84-0/)
* [Boost.Functional/Factory](https://www.stage.boost.cppalliance.org/libraries/functionalfactory/boost-1-84-0/)
* [**Boost.Functional/Overloaded Function**](https://www.stage.boost.cppalliance.org/libraries/functionaloverloaded_function/boost-1-84-0/)
* [**Boost.STL_interfaces**](https://www.stage.boost.cppalliance.org/libraries/stl_interfaces/boost-1-84-0/)
* [**Boost.Property Map (Parallel)**](https://www.stage.boost.cppalliance.org/libraries/property_map_parallel/boost-1-84-0/)

In fact all the cases marked in bold are wrong. Official [naming guidelines](https://www.boost.org/development/requirements.html#Naming_consistency) are woefully incomplete in regards to the many potential situations exemplified, stating only that a library named "xxx" is stylized as "Boost.Xxx". This doesn't cover patterns such as "xxx yyy", "xxx-yyy", etc. actually in use, so let's look at the established practice:

* "xxx yyy" goes to "Boost.XxxYyy": "Container Hash" --> "Boost.ContainerHash"
* "xxx-yyy" goes to "Boost.XxxYyy": "Multi-Array" --> "Boost.MultiArray"
* "xxx/yyy" goes to "Boost.Xxx/Yyy": The only cases here are Boost.Functional/* libraries, which internally have decided to keep the slash symbol (see [docs](https://www.stage.boost.cppalliance.org/doc/libs/1_84_0/libs/functional/factory/doc/html/index.html)).
* "xxx/yyy zzz" goes to "Boost.Xxx/YyyZzz", the only case being [Boost.Functional/OverloadedFunction](https://www.stage.boost.cppalliance.org/doc/libs/1_84_0/libs/functional/overloaded_function/doc/html/index.html).
* "xxx_yyy" goes to "Boost.XxxYyy", the only case being [Boost.STLInterfaces](https://www.stage.boost.cppalliance.org/doc/libs/1_84_0/doc/html/stl_interfaces.html).
* The only "xxx (yyy)" case is [Property Map (Parallel)](https://www.stage.boost.cppalliance.org/doc/libs/1_84_0/libs/property_map_parallel/index.html), which unfortunately uses internally the incorrect stylization ["Boost Property Map (Parallel)"](https://www.stage.boost.cppalliance.org/doc/libs/1_84_0/libs/property_map_parallel/index.html) (no dot used). My suggestion is that we use "Boost.PropertyMap(Parallel)" here.

To sum it up, the correct stylized names are:

* Boost.Accumulators
* Boost.ContainerHash
* Boost.MultiArray
* Boost.Functional/Factory
* Boost.Functional/OverloadedFunction
* Boost.STLInterfaces
* Boost.PropertyMap(Parallel)

and the associated styilization algorithm can be defined as follows:

* Split the library name as given in the Libraries [home page](https://www.stage.boost.cppalliance.org/libraries)  into *words* (alphanumeric characters, including digits) and *separators* (anything else, like blanks and "-")
* Capitalize the initial letter of each word
* Eliminate the following separators: white space, "-", "_"
* Prepend "Boost."

**Comment by kennethreitz** [comment]
These should all be better now on staging: https://www.stage.boost.cppalliance.org/libraries/msm/boost-1-84-0/

An exception is STLInterfaces, but its name in the database is `Stl_interfaces`. This originates from `libraries.json`.

**Comment by joaquintides** [comment]
Reopening because of these two cases:

* PolyCollection is styilized as Boost.Polycollection (should be Boost.PolyCollection)
* ThrowException is stylized as Boost.Throwexception (should be Boost.ThrowException)

In general, capitalization of the first letter should not touch other letters that may be capitalized.

**Comment by kennethreitz** [comment]
nice catch! fixed in staging now
